r/tests/testthat/test-schema.R

```diff
@@ -260,3 +260,22 @@ test_that("as_schema() works for StructType objects", {
   struct_type <- struct(col1 = int32())
   expect_equal(as_schema(struct_type), schema(col1 = int32()))
 })
+
+test_that("schema name assignment", {
+  schm <- schema(x = int8(), y = string(), z = double())
+  expect_identical(names(schm), c("x", "y", "z"))
+  names(schm) <- c("a", "b", "c")
+  expect_identical(names(schm), c("a", "b", "c"))
+  expect_error(names(schm) <- "f")
+  expect_error(names(schm) <- letters)
+  expect_error(names(schm) <- character(0))
+  expect_error(names(schm) <- NULL)
+  expect_error(names(schm) <- c(TRUE, FALSE))
```

Comment:
I agree with your point, but we also discourage testing of error messages which come from libarrow. I guess I can add some custom error handling that we can test for.

Reply:
L269-271 are all the same, right? strings but not the right length, so libarrow will raise.

L272-273 will raise from the cpp11 wrapping since you haven't passed strings. 

So maybe you can reduce this to two tests: one that looks for the "wrong number of args" message, and one that looks for the type mismatch message. Those seem safe enough to make assertions about, and minimizes the risk from asserting about other libraries' error messages.

This is another case that #34826 would help with.